client/components/data/query/index.jsx

```diff
@@ -0,0 +1,59 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { PureComponent } from 'react';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import { values } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import compareProps from 'lib/compare-props';
+import { getRequest } from 'state/selectors';
+
+class Query extends PureComponent {
+	static propTypes = {
+		compareOptions: PropTypes.object,
+		requestAction: PropTypes.func.isRequired,
+		requesting: PropTypes.bool.isRequired,
+	};
+
+	componentWillMount() {
+		this.comparator = compareProps( {
+			ignore: [ 'compareOptions', 'requestAction', 'requesting' ],
+			...this.props.compareOptions,
+		} );
+		this.request( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( ! this.comparator( this.props, nextProps ) ) {
+			this.request( nextProps );
+		}
+	}
+
+	// eslint-disable-next-line no-unused-vars
+	request( { compareOptions, requestAction, requesting, ...props } ) {
+		if ( requesting ) {
+			return;
+		}
+
+		requestAction( ...values( props ) );
```

Comment:
So the order how the props are specified in JSX markup does matter and is actually more important than the prop names? That's very unexpected and error prone.